Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs

```diff
@@ -1682,862 +1033,868 @@ protected internal bool connect()
+                    logger.Warning("Unsupported protocol version '{0}'. Use at your own risk.", protocol);
 #endif
-				}
-				if (reconnected)
-				{
+                }
+                if (reconnected)
+                {
 #if LOGGER
-					logger.Info("Send Challenge action.");
+                    logger.Info("Send Challenge action.");
 #endif
-					ChallengeAction challengeAction = new ChallengeAction();
-					try
-					{
-						SendAction(challengeAction, null);
-					}
+                    ChallengeAction challengeAction = new ChallengeAction();
+                    try
+                    {
+                        SendAction(challengeAction, null);
+                    }
 #if LOGGER
-					catch(Exception ex)
-					{
-						logger.Info("Send Challenge fail : ", ex.Message);
+                    catch (Exception ex)
+                    {
+                        logger.Info("Send Challenge fail : ", ex.Message);
 #else
 					catch
 					{
 #endif
-						disconnect(true);
-					}
-					return;
-				}
-			}
-			#endregion
-
-			if (reconnected && e is DisconnectEvent)
-			{
-				((DisconnectEvent)e).Reconnect = true;
-				fireEvent(e);
-				reconnect(false);
-			}
-			else if (!reconnected && reconnectEnable && (e is DisconnectEvent || e is ReloadEvent || e is ShutdownEvent))
-			{
-				((ConnectionStateEvent)e).Reconnect = true;
-				fireEvent(e);
-				reconnect(true);
-			}
-			else
-				fireEvent(e);
-		}
-
-		private void eventComplete(IAsyncResult result)
-		{
-		}
-
-		private void fireEvent(ManagerEvent e)
-		{
-			if (enableEvents && internalEvent != null)
-                if(UseASyncEvents)
-				    internalEvent.BeginInvoke(this, e, new AsyncCallback(eventComplete), null);
+                        disconnect(true);
+                    }
+                    return;
+                }
+            }
+            #endregion
+
+            if (reconnected && e is DisconnectEvent)
+            {
+                ((DisconnectEvent)e).Reconnect = true;
+                fireEvent(e);
+                reconnect(false);
+            }
+            else if (!reconnected && reconnectEnable && (e is DisconnectEvent || e is ReloadEvent || e is ShutdownEvent))
+            {
+                ((ConnectionStateEvent)e).Reconnect = true;
+                fireEvent(e);
+                reconnect(true);
+            }
+            else
+                fireEvent(e);
+        }
+
+        private void eventComplete(IAsyncResult result)
+        {
+        }
+
+        private void fireEvent(ManagerEvent e)
+        {
+            if (enableEvents && internalEvent != null)
+                if (UseASyncEvents)
+                    internalEvent.BeginInvoke(this, e, new AsyncCallback(eventComplete), null);
                 else
                     internalEvent.Invoke(this, e);
-		}
-		#endregion
-	}
+        }
+
+        private void fireEvent<T>(EventHandler<T> astEvent, ManagerEvent arg) where T : ManagerEvent

```

Comment:
I know we've not asked for it in the past, but we need to start getting inline docs on all new code, even private methods (help with code readability). Could we look to add some here to give reasons for new methods/overrides etc?